synapse/lib/stormlib/vault.py

```diff
@@ -483,36 +540,53 @@ async def setitem(self, name, valu):
         else:
             valu = await s_stormtypes.toprim(valu)
 
-        return await self.runt.snap.core.setVaultData(self.valu, name, valu)
+        return await self.runt.snap.core.setVaultConfigs(self.valu, name, valu)
 
     async def deref(self, name):
         vault = self.runt.snap.core.reqVault(self.valu)
 
         name = await s_stormtypes.tostr(name)
 
-        data = vault.get('data')
+        data = vault.get(self._vault_field_name)
         return data.get(name, None)
 
     async def iter(self):
         vault = self.runt.snap.core.reqVault(self.valu)
-        data = vault.get('data')
+        data = vault.get(self._vault_field_name)
 
         for item in data.items():
             yield item
 
     def __len__(self):
         vault = self.runt.snap.core.reqVault(self.valu)
-        data = vault.get('data')
+        data = vault.get(self._vault_field_name)
         return len(data)
 
     async def stormrepr(self):
         vault = self.runt.snap.core.reqVault(self.valu)
-        data = vault.get('data')
+        data = vault.get(self._vault_field_name)
         return repr(data)
 
     def value(self):
         vault = self.runt.snap.core.reqVault(self.valu)
-        return vault.get('data')
+        return vault.get(self._vault_field_name)
+
+class VaultSecrets(VaultConfigs):
+    _vault_field_name = 'secrets'
+    _vault_perm = s_cell.PERM_EDIT
+
+    @s_stormtypes.stormfunc(readonly=False)
```

Comment:
We can remove the `readonly=False`  decorators since that is the default behavior.